powerstore/datasource_protection_policy_test.go

```diff
@@ -0,0 +1,86 @@
+package powerstore
+
+import (
+	"os"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+// Test to Fetch Protection Policy
+func TestAccProtectionPolicy_FetchPolicy(t *testing.T) {
+	if os.Getenv("TF_ACC") == "" {
+		t.Skip("Dont run with units tests because it will try to create the context")
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: testProviderFactory,
+		Steps: []resource.TestStep{
+			{
+				Config: ProtectionPolicyDataSourceparamsName,
+			},
+			{
+				Config: ProtectionPolicyDataSourceparamsID,
+			},
+			{
+				Config: ProtectionPolicyDataSourceparamsAll,
+			},
+			{
+				Config:      ProtectionPolicyDataSourceparamsNameNegative,
+				ExpectError: regexp.MustCompile("nable to Read PowerStore Protection"),
+			},
+		},
+	})
+}
+
+var ProtectionPolicyDataSourceparamsName = `
+provider "powerstore" {
+	username = "` + username + `"
```

Comment:
This piece of code is repetitive. Is there a way to avoid it ??

Reply:
done